osquery/table/corefoundation_helpers_darwin.go

```diff
@@ -0,0 +1,83 @@
+package table
+
+/*
+#cgo darwin CFLAGS: -DDARWIN -x objective-c
+#include <CoreFoundation/CoreFoundation.h>
+*/
+import "C"
+import (
+	"fmt"
+	"reflect"
+	"unsafe"
+)
+
+func copyPreferenceValue(key, domain, username string) interface{} {
+	keyCFString := CFStringRef(key)
+	defer C.CFRelease((C.CFTypeRef)(keyCFString))
+	domainCFString := CFStringRef(domain)
+	defer C.CFRelease((C.CFTypeRef)(domainCFString))
+	usernameCFString := CFStringRef(username)
+	defer C.CFRelease((C.CFTypeRef)(usernameCFString))
+
+	val := C.CFPreferencesCopyValue(
+		keyCFString, domainCFString, usernameCFString, C.kCFPreferencesAnyHost,
+	)
+	defer C.CFRelease((C.CFTypeRef)(val))
+	return goValueFromCFPlistRef(val)
+}
+
+// CFStringRef returns a C.CFStringRef which must be released with C.CFRelease
+func CFStringRef(s string) C.CFStringRef {
```

Comment:
should this and the below functions be private?

Reply:
Yeah, no reason for them to be public, I copied them from my own gist i found somewhere.